plugins/nanospell/plugin.js

```diff
@@ -18,769 +18,782 @@
+			// beacuse of a special case around nested lists,
+			// non-root block nodes must also be excluded.
+			// the text content of ckeditor bookmarks must also be excluded
+			// or &nbsp; will be added throughout.
+			var path = new CKEDITOR.dom.elementPath( node, startNode );
+
+			return node.type == CKEDITOR.NODE_TEXT && // it is a text node
+				node.getLength() > 0 &&  // and it's not empty
+				( !node.isReadOnly() ) &&   // or read only
+				isNotBookmark(node) &&  // and isn't a fake bookmarking node
+				(path.block && path.block.equals(startNode)); // it's not nested in a deeper block from our root.
+		}
+
+		this.rootBlockTextNodeWalker = new CKEDITOR.dom.walker(range);
+		rootBlockTextNodeWalker.evaluator = isRootBlockTextNode;
+
+		var wordSeparatorRegex = /[.,"'?!;: \u0085\u00a0\u1680\u280e\u2028\u2029\u202f\u205f\u3000]/;
+
+		this.isWordSeparator = function (character) {
+			if (!character)
+				return true;
+			var code = character.charCodeAt(0);
+			return ( code >= 9 && code <= 0xd ) || ( code >= 0x2000 && code <= 0x200a ) || wordSeparatorRegex.test(character);
+		};
+
+		this.textNode = walker.next();
+		this.offset = 0;
+		this.origRange = range;
+
+		this._ = {
+			walker: rootBlockTextNodeWalker,
+			isWordSeparator: isWordSeparator
+		}
+	}
+
+	WordWalker.prototype = {
+		getOffsetToNextNonSeparator: function (text, startIndex) {
+			var i, length;
+			length = text.length;
+
+			for (i = startIndex + 1; i < length; i++) {
+				if (!this._.isWordSeparator(text[i])) {
+					break;
+				}
+			}
+
+			return i;
+
+		},
+		getNextWord: function () {
+
+			// iterate through each of the text nodes in the walker
+			// break, store current offset, and return a range when finding a word separator
+			// until all text nodes in the walker are exhausted.
+
+			var word = '';
+			var currentTextNode = this.textNode;
+			var wordRange = this.origRange.clone();
+			var i;
+			var text;
+
+			if (currentTextNode === null) {
+				return null;
+			}
+
+			wordRange.setStart(currentTextNode, this.offset);
+
+			while (currentTextNode !== null) {
+				text = currentTextNode.getText();
+				for (i = this.offset; i < text.length; i++) {
+					if (this._.isWordSeparator(text[i])) {
+						word += text.substr(this.offset, i - this.offset);
+						wordRange.setEnd(currentTextNode, i);
+
+						this.offset = this.getOffsetToNextNonSeparator(text, i);
+
+						return {
+							word: word,
+							range: wordRange
+						}
+					}
+				}
+				word += text.substr(this.offset);
+				this.offset = 0;
+				wordRange.setEndAfter(this.textNode);
+				currentTextNode = this._.walker.next();
+
+				this.textNode = currentTextNode;
+
+			}
+			// reached the end of paragraph
+
+			return {
+				word: word,
+				range: wordRange
+			};
+
+		}
+	};
+
+	CKEDITOR.plugins.add('nanospell', {
+		icons: 'nanospell',
+		init: function (editor) {
+			var self = this;
+
+			this.addRule(editor);
+
+			overrideCheckDirty();
+
+			settingsPath = this.path;
```

Comment:
Shouldn't this be `var settingsPath`? It's tough to tell in the diff with all of the nesting going on here.